### PR TITLE
Migrate datetime documentation to code

### PR DESCRIPTION
**Pull request by jatin510**
## Which issue does this PR close?
Closes https://github.com/apache/datafusion/issues/12859

Closes #.

## Rationale for this change

Documentation migration

## What changes are included in this PR?

Documentation related changes for datetime functions.

## Are these changes tested?

Yes.

### CODE REVIEW

**Comment by jonathanc-n** [comment]
LGTM, honestly the with_standard_argument isn't a necessary change, just enforces consistency.

**Comment by jatin510** [comment]
@jonathanc-n 
I checked and found:
Only two functions take an argument of a single data type:
`date_bin` - Time 
`from_unixtime` - Integer

But it is changing the meaning of the sentence

```
for `date_bin`
 #### Arguments
 
 - **interval**: Bin interval.
-- **expression**: Time expression to operate on. Can be a constant, column, or function.
+- **expression**: Time expression to operate on. Can be a constant, column, or function, and any combination of operators.
 - **origin-timestamp**: Optional. Starting point used to determine bin boundaries. If not specified defaults 1970-01-01T00:00:00Z (the UNIX epoch in UTC).
 
 The following intervals are supported:
@@ -1958,7 +1958,7 @@ from_unixtime(expression)
 
for `from_unixtime`
 #### Arguments
 
-- **expression**: Expression to operate on. Can be a constant, column, or function, and any combination of arithmetic operators.
+- **expression**: Integer expression to operate on. Can be a constant, column, or function, and any combination of operators.

```

So, I am not going with the use of  `with_standard_argument`.

File: docs/source/user-guide/sql/scalar_functions.md [hunk 1]
```diff
@@ -51,175 +51,7 @@ position(substr in origstr)
...
 
-- [now](#now)
-- [current_date](#current_date)
-- [current_time](#current_time)
-- [date_bin](#date_bin)
-- [date_trunc](#date_trunc)
-- [datetrunc](#datetrunc)
-- [date_part](#date_part)
-- [datepart](#datepart)
 - [extract](#extract)
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:54]: Is `extract` a special function? I think it might be as it has special syntax

**Review by alamb — APPROVED** [review]
Amazing -- thank you @jatin510  🙏

**Comment by alamb** [comment]
I merged up from main to resolve some conflicts with this PR

**Comment by jonathanc-n** [comment]
@jatin510 

.with_standard_argument('expression', None) is used for when it is just 'expression' at the beginning. In the example you showed that would be with_standard_argument("expression", Some("time"))

**Comment by jonathanc-n** [comment]
@jatin510 Oh, sorry about the confusion there, i got confused, the commit hasn't been pushed yet.

**Comment by alamb** [comment]
Thanks again @jatin510  and @jonathanc-n  -- sorry we can't merge these PRs faster. It is all we can do to keep up with reviews these days :)

**Pull request merged**